group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3984,9 +3985,9 @@ public CoordinatorResult<Void, CoordinatorRecord> classicGroupSync(
         SyncGroupRequestData request,
         CompletableFuture<SyncGroupResponseData> responseFuture
     ) throws UnknownMemberIdException, GroupIdNotFoundException {
-        Group group = groups.get(request.groupId(), Long.MAX_VALUE);
+        Group group = group(request.groupId());
```

Comment:
We need to catch GroupIdNotFoundException and re-thrown UnknownMemberIdException here.